core/src/tx.rs

```diff
@@ -85,19 +84,18 @@ impl TransactionValidator {
         wsv: &WorldStateView,
     ) -> Result<(), TransactionRejectionReason> {
         for tx in txs {
-            self.validate_internal(tx.into_v1(), true, wsv)?;
+            self.validate_internal::<true>(tx.into_v1(), wsv)?;
```

Comment:
I know the following isn't related to this PR, but, according to "true" here is hard-coded, it seems the current implementation implies that we should use this function only on genesis. I don't think this is a reliable assumption because nothing prevents to use it on non-genesis. I would add IS_GENESIS here and use the flag hard-coding above on the call-stack where we make sure the peer is genesis.

Reply:
that is a good observation